lib/modules/announcements.js

```diff
@@ -30,53 +31,70 @@ const recheckPostAfter = HOUR;
 
 const now = Date.now();
 
-module.go = () => {
+let post;
+let $biff;
+
+module.beforeLoad = async function() {
 	if (isCurrentSubreddit(subreddit)) {
 		setMarkedRead();
 	} else {
-		maybeNotify();
+		post = await postForNotification();
 	}
 };
 
-function openAnnouncements() {
-	location.href = viewUrl;
-}
+module.go = () => {
+	if (post) {
+		notify(post);
+	}
+};
 
-async function maybeNotify() {
+async function postForNotification() {
 	const post = await getLatestPost();
-	// has the user ever seen a notification? (i.e. is this a fresh install?)
-	const hasSeenNotification = await Storage.has(keys.lastUnreadDate);
-
 	if (!post) {
-		return;
+		return false;
 	}
+	cacheLatestPost(post);
 
-	// if the post is novel, mark the latest ID and unread date, even if we don't end up showing a notification
-	if (await Storage.get(keys.lastID) !== post.id) {
-		Storage.set(keys.lastID, post.id);
-		Storage.set(keys.lastUnreadDate, now);
-	}
+	const result = await shouldNotify(post);
+	return result ? post : false;
+}
 
+async function shouldNotify(post) {
 	if (post.created_js + archiveAfter <= now) {
 		// archived
-		return;
+		return false;
 	}
 
 	if (post.created_js <= await getMarkedRead()) {
 		// posted before last "marked read"
-		return;
+		return false;
 	}
 
+	return true;
+}
+
+const notify = _.once(async function(post) {
```

Comment:
Even if it shouldn't be called more than once because it's only called in  `go()`, if you're using it to uphold an invariant then :+1: